### PR TITLE
Authz api mnesia refactor

### PR DESCRIPTION
**Pull request by JimMoen**
use honconsc for authz-api-mnesia-refactor

### CODE REVIEW

File: apps/emqx_authz/src/emqx_authz_api_mnesia.erl [hunk 1]
```diff
@@ -642,6 +400,43 @@ format_by_clientid([{clientid, Clientid}, {rules, Rules}]) ->
 atom(B) when is_binary(B) ->
     try binary_to_existing_atom(B, utf8)
     catch
-        _ -> binary_to_atom(B)
+        '_' -> binary_to_atom(B)
```
Comment (zhongwencool) [inline apps/emqx_authz/src/emqx_authz_api_mnesia.erl:403]:
```erlang
catch:
  _Err:_Expection -> 
```
'_' is atom...

File: apps/emqx_authz/test/emqx_authz_api_mnesia_SUITE.erl [hunk 2]
```diff
@@ -126,65 +82,149 @@ set_special_configs(_App) ->
 %%------------------------------------------------------------------------------
 
 t_api(_) ->
-    {ok, 204, _} = request(post, uri(["authorization", "sources", "built-in-database", "username"]), [?EXAMPLE_USERNAME]),
-    {ok, 200, Request1} = request(get, uri(["authorization", "sources", "built-in-database", "username"]), []),
-    {ok, 200, Request2} = request(get, uri(["authorization", "sources", "built-in-database", "username", "user1"]), []),
+    {ok, 204, _} =
+        request( post
+               , uri(["authorization", "sources", "built-in-database", "username"])
+               , [?USERNAME_RULES_EXAMPLE]),
```
Comment (zhongwencool) [inline apps/emqx_authz/test/emqx_authz_api_mnesia_SUITE.erl:88]: Is that possible to use the swagger's example to do this test?
just like authn?
we can plan to do this in another PR.

**Review by zhongwencool — CHANGES_REQUESTED** [review]

File: apps/emqx_authz/src/emqx_authz_api_mnesia.erl (hunk 1 above)
Comment (JimMoen) [reply apps/emqx_authz/src/emqx_authz_api_mnesia.erl:403]: but elvis complained..

File: apps/emqx_authz/test/emqx_authz_api_mnesia_SUITE.erl (hunk 2 above)
Comment (JimMoen) [reply apps/emqx_authz/test/emqx_authz_api_mnesia_SUITE.erl:88]: `emqx_authz_api_mnesia.erl` and `emqx_authz_api_mnesia_SUITE.erl` use the same example in `emqx_authz.hrl`. 
They are provided to swagger, too.

Comment (zhongwencool) [reply apps/emqx_authz/test/emqx_authz_api_mnesia_SUITE.erl:88]: Great!

**Review by zhongwencool — APPROVED** [review]
